### ISSUE TITLE
[java] FN for LocalVariableCouldBeFinal

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3963]
**Affects PMD Version:6.45.0** 

**Rule:LocalVariableCouldBeFinal**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.45.0/pmd_rules_java_codestyle.html#localvariablecouldbefinal>

**Description:**
A false negative about the rule LocalVariableCouldBeFinal. PMD should have reported a warning in line 2, but it doesn't. However, this local variable is only assigned once, and it can be modified by `final`. Hence, I think this is a false negative.

**Code Sample demonstrating the issue:**
```java
public void foo() {
  int a;  // should report a warning in this line
  a = 0;
}
```

**Expected outcome:**

PMD should report a violation at line 2, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
